content/blog/2025/04/soft-ap-tutorial/index.md

```diff
@@ -0,0 +1,369 @@
+}
+```
+
+### Register handlers for soft-AP
+
+At this point, the function handling Wi-Fi events is as follows:
+
+```c
+static void wifi_event_handler(void* arg, esp_event_base_t event_base,
+                                  int32_t event_id, void* event_data){
+    printf("Event nr: %ld!\n", event_id);
+}
+```
+
+Now compile, flash, start a monitor and run the project. We should start seeing several event numbers appearing on the terminal.
+
+
+### Connect to the soft-AP with a Smartphone
+
+At this point, take your smartphone, open the Wi-Fi list, and select the SSID `esp_tutorial`. When we do so, we should see on the terminal `Event nr: 14!`. 
+ 
+{{< figure
+    default=true
+    src="img/ap_list.webp"
+    height=500
+    caption="Listing of APs"
+    >}}
+
+If we check on the `esp-idf` source code related to the event codes ([here](https://github.com/espressif/esp-idf/blob/c5865270b50529cd32353f588d8a917d89f3dba4/components/esp_wifi/include/esp_wifi_types_generic.h#L964)), we will find that 14 corresponds to `WIFI_EVENT_AP_STACONNECTED`. 
+
+This indicates that a station (STA) has connected to the soft-AP. 
+
+<!--  [!NOTE] 
+> You can use the `event_id`to discriminate between different events and then use the additional data you are provided with `event_data`.  
+ -->
+
+
+We can use the `event_id` to distinguish between different events and then process the additional data you available in `event_data`.  
+
+This allows us to handle specific Wi-Fi events appropriately, such as when a station connects or disconnects from the soft-AP. 
+
+
+{{< alert icon="lightbulb" iconColor="#179299"  cardColor="#9cccce">}}
+
+`event_data` is a void pointer, you need to cast it to the proper structure if you want to extract data. [Here](https://github.com/espressif/esp-idf/blob/c5865270b50529cd32353f588d8a917d89f3dba4/examples/wifi/getting_started/softAP/main/softap_example_main.c#L36) you can find two examples. 
+
+{{< /alert >}}
+
+You can find the complete code at the link [soft-ap tutorial code](https://github.com/FBEZ-docs-and-templates/devrel-tutorials-code/blob/main/tutorial-soft-ap/main/basic_soft_ap.c). 
+
+
+<!-- Note that `event_data` is a void pointer, you need to cast it to the proper structure if you want to extract data. [Here](https://github.com/espressif/esp-idf/blob/c5865270b50529cd32353f588d8a917d89f3dba4/examples/wifi/getting_started/softAP/main/softap_example_main.c#L36) you can find two examples. -->
+
+## Conclusion  
+
+In this tutorial, you learned how to:  
+
+1. Create a new project based on the `hello_world` example.  
+2. Initialize the `esp_netif` library.  
+3. Start the standard event loop to handle Wi-Fi events.  
+4. Configure and launch a soft-AP, allowing devices to connect via Wi-Fi.  
+5. Verify event handling by monitoring the terminal for connection events.  
+
+This serves as a foundation for building more advanced Wi-Fi applications, such as MQTT clients, HTTP servers, or other networked solutions.
```

Comment:
Consider adding the [ESP32-C6 workshop](https://preview-developer.espressif.com/pr404/workshops/esp-idf-with-esp32-c6/) as additional material.